### PR TITLE
[MaskRCNN] ONNX Dynamic Flatten

### PR DESCRIPTION
**Pull request by arogowie-intel**

### CODE REVIEW

File: src/ngraph/builder/reshape.cpp [hunk 1]
```diff
@@ -203,20 +206,93 @@ shared_ptr<Node> builder::opset1::transpose(const Output<Node>& value)
...
+        return builder::opset1::reshape(value, Shape{first_dim_size, last_dim_size});
+    }
+    else
+    {
+        shared_ptr<Node> output_shape;
+        if (axis == 0)
+        {
+            output_shape = ngraph::opset1::Constant::create(element::i64, Shape{2}, {1, -1});
+        }
+        else
```
Comment (jane-intel) [inline src/ngraph/builder/reshape.cpp:270]: Please apply change list from https://github.com/NervanaSystems/ngraph/pull/4505 here

Comment (arogowie-intel) [reply src/ngraph/builder/reshape.cpp:274]: Done.

File: src/ngraph/builder/reshape.cpp [hunk 2]
```diff
@@ -201,20 +204,97 @@ shared_ptr<Node> builder::opset1::transpose(const Output<Node>& value)
...
+            } // opset1
+        }     // builder
+    }         // ngraph
+}
 
-    size_t last_dim_size =
-        accumulate(next(begin(data_shape), axis), end(data_shape), 1UL, multiplies<size_t>());
+shared_ptr<Node> builder::opset1::flatten(const Output<Node>& value, int axis)
+{
+    if (value.get_partial_shape().is_static())
```
Comment (jane-intel) [inline src/ngraph/builder/reshape.cpp:246]: Please get rid of this branch at all, it breaks reshape-ability of the model (setting new Parameter shapes and propagating them down the function).

Comment (arogowie-intel) [reply src/ngraph/builder/reshape.cpp:246]: Unfortunately at this moment removing this branch breaks our current support for ONNX.

**Comment by arogowie-intel** [comment]
Removed Michał from reviewers as he's on sick leave.

**Pull request merged**